### PR TITLE
Added pd.DataFrame transformation for Imputer and HampelFilter

### PR DESCRIPTION
**Pull request by aiwalter**
#### What does this implement/fix? Explain your changes.
Added `pd.DataFrame` transformation for `Imputer` and `HampelFilter`.

___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

### CODE REVIEW

File: sktime/transformations/series/impute.py [hunk 1]
```diff
@@ -70,49 +70,70 @@ def transform(self, Z, X=None):
...
+        Z : pd.Series, pd.DataFrame
+            Transformed time series(es).
         """
         self.check_is_fitted()
         self._check_method()
-        z = check_series(Z, enforce_univariate=True)
+        Z = check_series(Z)
+        # multivariate
+        if isinstance(Z, pd.DataFrame):
+            for col in Z:
```
Comment (mloning) [inline sktime/transformations/series/impute.py:85]: Just a quick comment: looping over columns doesn't seem like the right approach, it should be possible to make use of more efficient methods with `pd.DataFrames` or the underlying `np.ndarray`s, e.g. the `.fillna(...)` method seems to work on data frames too.

Comment (aiwalter) [reply sktime/transformations/series/impute.py]: yes this is true, just added it whenever applicable.

**Review by mloning — COMMENTED** [review]
This looks good in general, what about testing?

**Comment by aiwalter** [comment]
should be tested like before? I have removed the tag `univariate-only` so I think the following existing test function is applied to them now?

https://github.com/alan-turing-institute/sktime/blob/7343dc36566935b42167878c45d570099af96dab/sktime/transformations/tests/test_all_transformers.py#L90-L100

___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

**Review by mloning — APPROVED** [review]
Okay, make sense for now. Eventually we should add some tests for edge-cases (e.g. all rows are nan) and numerical results.
